wire-ios-data-model/Source/Utilis/AutoMockable.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// Any protocol that extends `AutoMockable` will have a mock automatically generated
+/// by the Sourcery codegen tool.
+
+public protocol AutoMockable {}
```

Comment:
Hm... I know there is no good or bad answer here, but maybe for mocking, it's better to use annotations instead of protocol conformance?  I used both in the past and I must say that protocols were sometimes a bit confusing for developers - since it's a protocol that is used only to mark some type for codegen tool

Reply:
Sure, I added it here as an option, but we're also free to use the annotation. I never really used sourcery so I don't know what is the preferred approach, if you think only annotations, then I can remove this protocol.

Reply:
Yeap, I thought about not enabling the 'protocol' option to avoid confusion 😄

Reply:
ok I'll remove the protocol

Reply:
done